src/EditorFeatures/Core/Implementation/AddImports/CSharpAddImportsPasteCommandHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Commanding;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.AddImports
+{
+    [Export]
+    [Export(typeof(ICommandHandler))]
+    [ContentType(ContentTypeNames.CSharpContentType)]
+    [Name(PredefinedCommandHandlerNames.AddImportsPaste)]
+    // Order is important here, this command needs to execute before PasteTracking
+    // since it may modify the pasted span. Paste tracking dismisses if 
+    // the span is modified. It doesn't need to be before FormatDocument, but
+    // this helps the order of execution be more constant in case there 
+    // are problems that arise. This command will always execute the next
+    // command before doing operations.
+    [Order(After = PredefinedCommandHandlerNames.PasteTrackingPaste)]
+    [Order(Before = PredefinedCommandHandlerNames.FormatDocument)]
+    internal class CSharpAddImportsPasteCommandHandler : AbstractAddImportsPasteCommandHandler
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpAddImportsPasteCommandHandler(IThreadingContext threadingContext) : base(threadingContext)
+        {
+        }
+
+        public override string DisplayName => EditorFeaturesResources.Add_Missing_Usings_On_Paste;
```

Comment:
'on' should never be capitalized.  This can either be title case of `Add Missing Usings on Paste`.  Or sentence cases of `Add missing 'usings' on paste`.  I would prefer the latter.  Title case normally feels very weird unless it's the header of some section.

Reply:
Title casing makes sense here because it's the title of the command handler. There's a separation of the text being shown which will be title casing. "DisplayName" fulfills the interface contract, so unfortunately can't really label it more clearly. I'll add a comment though.